src/OAuthMessageHandler.cs

```diff
@@ -13,13 +13,27 @@ public class OAuthMessageHandler : DelegatingHandler
         private readonly string _secret;
         private readonly string _authToken;
         private readonly string _authTokenSecret;
+        private readonly string _versionParameter;
 
-        public OAuthMessageHandler(string apiKey, string secret, string authToken, string authTokenSecret)
+        public OAuthMessageHandler(string apiKey, string secret, string authToken, string authTokenSecret, OAuthVersionParameter versionParameter = OAuthVersionParameter.OneZeroA)
         {
             _apiKey = apiKey;
             _secret = secret;
             _authToken = authToken;
             _authTokenSecret = authTokenSecret;
+            switch (versionParameter)
```

Comment:
This code is duplicated. I suggest creating a helper method that takes an `OAuthVersion` enum and returns the string representation for it.